### PR TITLE
New exception handling diagnostic fixes

### PR DESCRIPTION
**Pull request by janvorli**
This change makes the new exception handling pass all the mdbg tests (tested on Windows x64, Linux x64 and Linux arm64) except for the Exceptions.ReflectionTest that will require a change in the test. 
Here is a list of changes:
* Adds debugger and profiler events in the right places and order
* Adds debugger exception interception implementation
* Ensures that the new managed exception handling frames are not visible to the debugger stack trace
* Fixes some missing frames on stack traces (coming from `HelperMethodFrame`)
* Moves first chance exception handling to the native code. It is necessary to ensure that the exception stack trace contains frames from the explicit frames that have `MethodDesc` at the time the event fires.
* Makes some refactoring around the `ExInfo` - the `CONTEXT` and `REGDISPLAY` are now in the `ExInfo`, the `ExInfo` now also contains the original exception pointers.
* Fixes one case when an unhandled exception at the boundary of managed and native code was not handled correctly.
* Changes the way an unhandled exception in exception filter is swallowed. The stack frame iterator was having issues with the fact that the exception "leaked" from the actual filter and was caught by another filter in the managed EH. Now the exception is always swallowed at the filter.

### CODE REVIEW

**Comment by tommcdon** [comment]
@mikem8361

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs [hunk 1]
```diff
@@ -957,19 +1023,23 @@ private static void DebugVerifyHandlingFrame(UIntPtr handlingFrameSP)
...
 #else
                             InternalCalls.RhpCallFilterFunclet(ObjectHandleOnStack.Create(ref exception), pFilterFunclet, frameIter.RegisterSet);
 #endif
+#if NATIVEAOT
                     }
                     catch when (true)
                     {
                         // Prevent leaking any exception from the filter funclet
                     }
+#endif
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs:1042]:
```suggestion
#if NATIVEAOT
                    try
                    {
                        shouldInvokeHandler = InternalCalls.RhpCallFilterFunclet(exception, pFilterFunclet, frameIter.RegisterSet);
                    }
                    catch when (true)
                    {
                        // Prevent leaking any exception from the filter funclet
                    }
#else // NATIVEAOT
                    InternalCalls.RhpCallFilterFunclet(ObjectHandleOnStack.Create(ref exception), pFilterFunclet, frameIter.RegisterSet);
#endif // NATIVEAOT
```

File: src/libraries/System.Private.CoreLib/src/System/AppContext.cs [hunk 2]
```diff
@@ -82,12 +82,7 @@ public static void SetData(string name, object? data)
         internal static event EventHandler<FirstChanceExceptionEventArgs>? FirstChanceException;
 #pragma warning restore CS0067
 
-#if !NATIVEAOT
-        internal static void OnFirstChanceException(object e)
-        {
-            FirstChanceException?.Invoke(AppDomain.CurrentDomain, new FirstChanceExceptionEventArgs((Exception)e));
-        }
-#endif
+        internal static event EventHandler? ProcessExit;
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/AppContext.cs:85]: Build break - incorrectly resolved merge conflict?

Comment (janvorli) [reply src/libraries/System.Private.CoreLib/src/System/AppContext.cs]: Yes, fixed

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs (hunk 1 above)
Comment (janvorli) [reply src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs:1042]: Right, it looks better. I will update it this way.

File: src/coreclr/vm/exceptionhandling.cpp [hunk 3]
```diff
@@ -2300,6 +2306,46 @@ bool ExceptionTracker::HandleNestedExceptionEscape(StackFrame sf, bool fIsFirstP
...
+BOOL NotifyDebuggerOfStub(Thread* pThread, Frame* pCurrentFrame)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    BOOL fDeliveredFirstChanceNotification = FALSE;
+
+    // <TODO>
+    // Remove this once SIS is fully enabled.
+    // </TODO>
+    extern bool g_EnableSIS;
```
Comment (jkotas) [inline src/coreclr/vm/exceptionhandling.cpp:2319]: Delete this from here and everywhere else? It seems to be unconditionally enabled, so this TODO is no longer relevant.

File: src/coreclr/vm/exceptionhandling.cpp [hunk 4]
```diff
@@ -2300,6 +2306,46 @@ bool ExceptionTracker::HandleNestedExceptionEscape(StackFrame sf, bool fIsFirstP
...
+    {
+        _ASSERTE(GetThreadNULLOk() == pThread);
+
+        GCX_COOP();
+
+        // For debugger, we may want to notify 1st chance exceptions if they're coming out of a stub.
+        // We recognize stubs as Frames with a M2U transition type. The debugger's stackwalker also
+        // recognizes these frames and publishes ICorDebugInternalFrames in the stackwalk. It's
+        // important to use pFrame as the stack address so that the Exception callback matches up
+        // w/ the ICorDebugInternlFrame stack range.
```
Comment (jkotas) [inline src/coreclr/vm/exceptionhandling.cpp:2331]:
```suggestion
        // w/ the ICorDebugInternalFrame stack range.
```
Pre-existing typo (please fix other instances too)

**Review by jkotas — APPROVED** [review]

File: src/coreclr/vm/exceptionhandling.cpp (hunk 4 above)
Comment (janvorli) [reply src/coreclr/vm/exceptionhandling.cpp]: Fixed

File: src/coreclr/vm/exceptionhandling.cpp (hunk 3 above)
Comment (janvorli) [reply src/coreclr/vm/exceptionhandling.cpp]: Deleted

**Review by tommcdon — APPROVED** [review]
LGTM, thanks!

**Comment by janvorli** [comment]
The windows arm64 nativeaot leg failure is a known issue

**Pull request merged**